Jenkinsfile

```diff
@@ -39,6 +50,35 @@ pipeline {
         }
       }
     }
+    stage('Maven Build Branch') {
+      when {
+        not {
+          branch 'master'
+        }
+      }
+      steps {
+        script {
+          def stepBuildTime = System.currentTimeMillis()
+          def commit = gitCommit()
+          def branchName = gitBranchName()
+
+          withCredentials([
+                  string(credentialsId: 'pact_broker_username', variable: 'PACT_BROKER_USERNAME'),
+                  string(credentialsId: 'pact_broker_password', variable: 'PACT_BROKER_PASSWORD')]
+          ) {
+              sh 'mvn -version'
+              sh "mvn clean package pact:publish -DrunContractTests -DPACT_BROKER_URL=https://pact-broker-test.cloudapps.digital -DPACT_CONSUMER_VERSION=${commit}" +
```

Comment:
It seems the only difference between master and branch build is `runContractTests` parameter.
Is it not possible to aggregate them into one?

Reply:
Good point, but as it stands achieving this seemingly simple task is weirdly hard. It should be done, but I suggest we merge this as is to unblock further improvements of pact stuff